timelock-impl/src/test/java/com/palantir/atlasdb/timelock/paxos/PaxosStateLogMigrationIntegrationTest.java

```diff
@@ -0,0 +1,236 @@
+
+        resetPaxosComponents();
+        PaxosLearner learner = paxosComponents.learner(CLIENT);
+
+        assertValuePresent(secondRound, sqliteLog);
+        assertValueLearned(secondRound, learner);
+    }
+
+    @Test
+    public void noCrossClientPollution() throws IOException {
+        int round = 200;
+        int otherRound = 100;
+        PaxosStateLog<PaxosValue> fileBasedLog = createFileSystemLog(CLIENT);
+        fileBasedLog.writeRound(round, valueForRound(round));
+
+        PaxosLearner learner = paxosComponents.learner(CLIENT);
+
+        Client otherClient = Client.of("other");
+        PaxosStateLog<PaxosValue> otherFileBasedLog = createFileSystemLog(otherClient);
+        otherFileBasedLog.writeRound(otherRound, valueForRound(otherRound));
+        fileBasedLog.writeRound(round, valueForRound(round));
+
+        PaxosLearner otherLearner = paxosComponents.learner(otherClient);
+        PaxosStorageParameters otherParameters = getParametersForClient(otherClient);
+        PaxosStateLog<PaxosValue> otherSqliteLog = createSqliteLog(otherParameters);
+
+        assertValueAbsent(round, otherSqliteLog);
+        assertValuePresent(otherRound, otherSqliteLog);
+        assertValueLearned(round, learner);
+        assertValueNotLearned(otherRound, learner);
+        assertValueNotLearned(round, otherLearner);
+        assertValueLearned(otherRound, otherLearner);
+    }
+
+    private void assertValueLearned(int secondRound, PaxosLearner learner) {
+        assertThat(learner.getLearnedValue(secondRound)).hasValue(valueForRound(secondRound));
+    }
+
+    private void assertValueNotLearned(int rogueValue, PaxosLearner learner) {
+        assertThat(learner.getLearnedValue(rogueValue)).isEmpty();
+    }
+
+    private void assertValuePresent(int rogueValue, PaxosStateLog<PaxosValue> sqliteLog) throws IOException {
+        assertThat(PaxosValue.BYTES_HYDRATOR.hydrateFromBytes(sqliteLog.readRound(rogueValue)))
+                .isEqualTo(valueForRound(rogueValue));
+    }
+
+    private void assertValueAbsent(int nonMigratedRound, PaxosStateLog<PaxosValue> sqliteLog)
+            throws IOException {
+        assertThat(sqliteLog.readRound(nonMigratedRound)).isNull();
+    }
+
+    private void resetPaxosComponents() {
+        paxosComponents = new LocalPaxosComponents(
+                TimelockPaxosMetrics.of(useCase, MetricsManagers.createForTests()),
+                useCase,
+                legacyDirectory,
+                sqliteDirectory, UUID.randomUUID(), true);
+    }
+
+    private PaxosValue valueForRound(int i) {
+        return new PaxosValue("value", i, new byte[] {1});
+    }
+
+    private PaxosStateLog<PaxosValue> createFileSystemLog(Client client) {
+        Path dir = useCase.logDirectoryRelativeToDataDirectory(legacyDirectory).resolve(client.value());
+        String learnerLogDir = Paths.get(dir.toString(), PaxosTimeLockConstants.LEARNER_SUBDIRECTORY_PATH).toString();
```

Comment:
nit: `dir.resolve(PaxosTimeLockConstants.LEARNER_SUBDIRECTORY_PATH).toString()`?

Reply:
I am pretty sure you are correct, but am not about to refactor and cause massive data corruption :P